sklearn/decomposition/_fastica.py

```diff
@@ -547,9 +561,23 @@ def g(x, fun_args):
             XT -= X_mean[:, np.newaxis]
 
             # Whitening and preprocessing by PCA
-            u, d, _ = linalg.svd(XT, full_matrices=False, check_finite=False)
+            if self.svd_solver == "eigh":
+                # Faster when num_samples >> n_features
+                D, u = linalg.eigh(X.T.dot(X))
```

Comment:
We likely need to adjust the order of `D` and `u` to match with svd.

[eigh](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.eigh.html) returns the eighvalues in ascending order, which [svd](https://docs.scipy.org/doc/scipy/reference/generated/scipy.linalg.svd.html) return it in non-decreasing order.